packages/react-core/src/components/Form/__tests__/FormContext.test.tsx

```diff
@@ -0,0 +1,261 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { FormContextConsumer, FormContextProvider } from '../FormContext';
+
+const customRender = (ui, initialValues?: Record<string, any>) => {
+  return render(<FormContextProvider initialValues={initialValues}>{ui}</FormContextProvider>);
+};
+
+describe('FormContext', () => {
+  test('consumers can show all values from "values" provided by "initialValues"', () => {
+    const initialValues = { fieldOneId: 'some value', fieldTwoId: 'some other value' };
+
+    customRender(
+      <FormContextConsumer>
+        {({ values }) => Object.values(values).map((value) => <span key={value}>{value}</span>)}
+      </FormContextConsumer>,
+      initialValues
+    );
+
+    Object.values(initialValues).forEach((initialValue) => expect(screen.getByText(initialValue)).toBeVisible());
+  });
+
+  test('consumer results in "isValid" as "true" without errors and "false" with errors', async () => {
+    const user = userEvent.setup();
+    const initialValues = { someId: 'some value' };
+    const onChange = (setValue, setError) => (event) => {
+      const { value } = event.target;
+      setValue('someId', value);
+
+      if (!value) {
+        setError('someId', 'Some error!');
+      }
+    };
+
+    customRender(
+      <FormContextConsumer>
+        {({ values, setValue, setError, isValid }) => (
+          <>
+            <input onChange={onChange(setValue, setError)} value={values.someId} />
+            {`isValid: ${isValid}`}
+          </>
+        )}
+      </FormContextConsumer>,
+      initialValues
+    );
+
+    expect(screen.getByText(/^isValid:/)).toHaveTextContent('isValid: true');
+
+    await user.clear(screen.getByRole('textbox'));
+    expect(screen.getByText(/^isValid:/)).toHaveTextContent('isValid: false');
+  });
+
+  test('using getValue, the consumer can show a value passed from initialValues', () => {
+    customRender(<FormContextConsumer>{({ getValue }) => <>{getValue('someId')}</>}</FormContextConsumer>, {
+      someId: 'some value'
+    });
+
+    expect(screen.getByText('some value')).toBeVisible();
+  });
+
+  test('using getValue, the consumer can show a value updated using setValue', () => {
+    customRender(
+      <FormContextConsumer>
+        {({ getValue, setValue }) => (
+          <>
+            <input onChange={(event) => setValue('someId', event.target.value)} />
+            <span>{getValue('someId')}</span>
+          </>
+        )}
+      </FormContextConsumer>,
+      {
+        someId: 'some value'
+      }
+    );
+
+    expect(screen.getByText('some value')).toBeVisible();
+  });
```

Comment:
I might just be misunderstanding this test, but shouldn't the input need to be changed?

Reply:
I think that was the intention, yes. It looks like the test just below this does exactly that, so I'll just remove this one.